src/test/groovy/edu/ie3/datamodel/io/factory/timeseries/IconTimeBasedWeatherValueFactoryTest.groovy

```diff
@@ -115,15 +115,15 @@ class IconTimeBasedWeatherValueFactoryTest extends Specification {
 		actual.with {
 			assert it.time == TimeUtil.withDefaults.toZonedDateTime("2019-08-01 01:00:00")
 			assert it.value.coordinate == coordinate
-			assert it.value.irradiation.directIrradiation.isPresent()
-			assert it.value.irradiation.directIrradiation.get() == Quantities.getQuantity(0.002317613203124999, PowerSystemUnits.KILOWATT_PER_SQUAREMETRE)
-			assert it.value.irradiation.diffuseIrradiation.isPresent()
-			assert it.value.irradiation.diffuseIrradiation.get() == Quantities.getQuantity(0.0018088226191406245, PowerSystemUnits.KILOWATT_PER_SQUAREMETRE)
-			assert it.value.temperature.temperature.isPresent()
+			assert it.value.solarIrradiance.directIrradiance.present
+			assert it.value.solarIrradiance.directIrradiance.get() == Quantities.getQuantity(0.002317613203124999, PowerSystemUnits.KILOWATT_PER_SQUAREMETRE)
```

Comment:
I would have expected this... :thinking: 
```suggestion
			assert it.value.solarIrradiance.directIrradiance.get() == Quantities.getQuantity(2.317613203124999, PowerSystemUnits.WATT_PER_SQUAREMETRE)
```

Reply:
groovy `==` first checks `compareTo` and uses `a.comporeTo(b) == 0` if a `compareTo` method exists. Otherwise it will use `equals`. As  a `compareTo` method exists for the quantity implementation, it is called first. This `compareTo` converts units if necessary and `0.002317613203124999 kW/m^2 = 2.317613203124999 W/m^2`. Hence the assert is `true`.